### PR TITLE
hive: Bump golang version to 1.22, OCP to 4.17

### PR DESCRIPTION
**Pull request by lleshchi**

### CODE REVIEW

File: ci-operator/config/openshift/hive/openshift-hive-master.yaml [hunk 1]
```diff
@@ -19,14 +19,14 @@ build_root:
     tag: rhel-9-release-golang-1.21-openshift-4.16
```
Comment (dlom) [inline ci-operator/config/openshift/hive/openshift-hive-master.yaml:19]: should this be 4.17? I don't know enough about this yaml file

File: ci-operator/config/openshift/hive/openshift-hive-master.yaml [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
 - dockerfile_path: Dockerfile
-  from: ocp_4.16_base-rhel9
+  from: ocp_4.17_base-rhel9
   inputs:
-    ocp_builder_rhel-8-golang-1.21-openshift-4.16:
+    ocp_builder_rhel-8-golang-1.22-openshift-4.17:
       as:
-      - registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.21-openshift-4.16
-    openshift_release_rhel-9-release-golang-1.21-openshift-4.16:
+      - registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17
```
Comment (2uasimojo) [inline ci-operator/config/openshift/hive/openshift-hive-master.yaml:26]: Is there a rhel-8-golang-1.22-openshift-4.17 builder?

If not, we'll have to find a rhel8/golang1.22 builder _somewhere_...

**Comment by 2uasimojo** [comment]
/pj-rehearse periodic-ci-openshift-hive-master-periodic-e2e-gcp-weekly periodic-ci-openshift-hive-master-periodic-e2e-weekly

Hit at least a couple of representative periodics too.

**Comment by lleshchi** [comment]
/pj-rehearse periodic-ci-openshift-hive-master-periodic-e2e-gcp-weekly

**Comment by lleshchi** [comment]
/pj-rehearse periodic-ci-openshift-hive-master-periodic-e2e-gcp-weekly

**Pull request merged**